app/modules/encounters/models.py

```diff
@@ -104,12 +104,26 @@ def add_assets_no_context(self, asset_list):
         for asset in asset_list:
             self.add_asset_no_context(asset)
 
+    # we dont delete .assets here because that is complex (due to annotations)
     def delete(self):
         with db.session.begin():
             while self.assets:
+                # this is actually removing the EncounterAssets joining object (not the assets)
                 db.session.delete(self.assets.pop())
             db.session.delete(self)
 
+    def delete_cascade(self):
+        assets = self.get_assets()
+        # TODO modify behavior when we have annotations
```

Comment:
I'm thinking here and the  delete is where the permissions owner_or_privilidged() method should be used to determine permission to do this